src/compiler/execution.jl

```diff
@@ -190,16 +190,16 @@ end
 
 ## kernel launching and argument encoding
 
-function (kernel::HostKernel)(args...; grid=1, threads=1, queue=global_queue(current_device()))
-    grid = MTLSize(grid)
+function (kernel::HostKernel)(args...; groups=1, threads=1, queue=global_queue(current_device()))
+    groups = MTLSize(groups)
     threads = MTLSize(threads)
-    (grid.width>0 && grid.height>0 && grid.depth>0) ||
-        throw(ArgumentError("Grid dimensions should be non-null"))
+    (groups.width>0 && groups.height>0 && groups.depth>0) ||
+        throw(ArgumentError("All group dimensions should be non-zero"))
     (threads.width>0 && threads.height>0 && threads.depth>0) ||
-        throw(ArgumentError("Threadgroup dimensions should be non-null"))
+        throw(ArgumentError("All thread dimensions should be non-zero"))
 
     (threads.width * threads.height * threads.depth) > kernel.pipeline_state.maxTotalThreadsPerThreadgroup &&
-        throw(ArgumentError("Max total threadgroup size should not exceed $(kernel.pipeline_state.maxTotalThreadsPerThreadgroup)"))
+        throw(ArgumentError("Maximum number of threads in a group should not exceed $(kernel.pipeline_state.maxTotalThreadsPerThreadgroup)"))
```

Comment:
```suggestion
        throw(ArgumentError("Number of threads in group ($(threads.width * threads.height * threads.depth)) should not exceed $(kernel.pipeline_state.maxTotalThreadsPerThreadgroup)"))
```
I've encountered this error in 1.9 and I think this makes the error message a bit more helpful.